src/metabase/driver/postgres.clj

```diff
@@ -813,6 +813,31 @@
                          (StringReader.))]
            (.copyIn copy-manager ^String sql tsvs)))))))
 
+(defmethod driver/current-user-table-privileges :postgres
+  [_driver database]
+  (let [conn-spec (sql-jdbc.conn/db->pooled-connection-spec database)]
+    (jdbc/query
+     conn-spec
+     (str/join
+        "\n"
+        ["with table_privileges as ("
+         "select"
+         "  true as is_current_user,"
+         "  current_user as role,"
+         "  t.schemaname as schema,"
+         "  t.tablename as table,"
+         "  pg_catalog.has_table_privilege(current_user, t.schemaname || '.' || t.tablename, 'SELECT') as select,"
+         "  pg_catalog.has_table_privilege(current_user, t.schemaname || '.' || t.tablename, 'UPDATE') as update,"
+         "  pg_catalog.has_table_privilege(current_user, t.schemaname || '.' || t.tablename, 'INSERT') as insert,"
+         "  pg_catalog.has_table_privilege(current_user, t.schemaname || '.' || t.tablename, 'DELETE') as delete"
+         "from pg_catalog.pg_tables t"
+         "where t.schemaname not in ('pg_toast', 'pg_catalog', 'information_schema')"
```

Comment:
is this list exhaustive? 

Maybe it's worth checking by prefix `pg_` since it's preserved anw.

> Schema names beginning with pg_ are reserved for system purposes and cannot be created by users.

https://www.postgresql.org/docs/current/ddl-schemas.html

Reply:
Ah great. 46278cf248caf8736a0e85ddd6b9d4c481c6c419